utils/common/src/main/java/org/apache/brooklyn/config/ConfigInheritance.java

```diff
@@ -20,31 +20,60 @@
 
 import java.io.Serializable;
 
+import org.apache.brooklyn.util.text.Strings;
+
 import com.google.common.annotations.Beta;
 
 @SuppressWarnings("serial")
 public abstract class ConfigInheritance implements Serializable {
 
-    public static final ConfigInheritance ALWAYS = new Always();
+    public enum InheritanceMode {
```

Comment:
Agree they represent the same thing currently. But on the other hand `ConfigInheritance` allows us to have context-dependent behaviour. We didn't need it so far, but doesn't hurt leaving it "just in case".